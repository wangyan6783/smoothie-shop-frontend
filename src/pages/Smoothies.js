import React, { useState, useEffect } from "react";
import Smoothie from "../components/Smoothie";
import SmoothieModal from "../components/SmoothieModal";

function Smoothies() {
  const [smoothies, setSmoothies] = useState([]);
  const [selectedSmoothie, setSelectedSmoothie] = useState(null);

  useEffect(fetchSmoothies, []);

  function fetchSmoothies() {
    fetch("http://localhost:5000/smoothies")
      .then(r => r.json())
      .then(smoothies => setSmoothies(smoothies));
  }

  function updateSmoothie(smoothie) {
    setSelectedSmoothie(smoothie);
  }

  return (
    <React.Fragment>
      <div className="smoothies-list">
        {smoothies.map(smoothie => (
          <Smoothie
            key={smoothie._id}
            smoothie={smoothie}
            updateSmoothie={updateSmoothie}
          />
        ))}
      </div>
      {selectedSmoothie && (
        <SmoothieModal
          smoothie={selectedSmoothie}
          updateSmoothie={updateSmoothie}
        />
      )}
    </React.Fragment>
  );
}

export default Smoothies;
