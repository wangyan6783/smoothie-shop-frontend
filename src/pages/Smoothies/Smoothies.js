import React, { useState } from "react";
import { useFetch } from "../../hooks/useFetch";
import Smoothie from "../../components/Smoothie/Smoothie";
import SmoothieModal from "../../components/SmoothieModal/SmoothieModal";

function Smoothies() {
  const [selectedSmoothie, setSelectedSmoothie] = useState(null);
  const { isLoading, fetchedData } = useFetch(
    "http://localhost:5000/smoothies"
  );
  const smoothies = fetchedData ? fetchedData : [];

  function updateSmoothie(smoothie) {
    setSelectedSmoothie(smoothie);
  }

  return (
    <React.Fragment>
      <div className="smoothies-list">
        {isLoading ? (
          <div>Loading...</div>
        ) : (
          smoothies.map(smoothie => (
            <Smoothie
              key={smoothie._id}
              smoothie={smoothie}
              updateSmoothie={updateSmoothie}
            />
          ))
        )}
      </div>
      {selectedSmoothie && (
        <SmoothieModal
          smoothie={selectedSmoothie}
          updateSmoothie={updateSmoothie}
        />
      )}
    </React.Fragment>
  );
}

export default Smoothies;
