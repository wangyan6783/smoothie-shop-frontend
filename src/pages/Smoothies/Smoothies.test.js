import React from "react";
import renderer from "react-test-renderer";
import Smoothies from "./Smoothies";

describe("Smoothies Component", () => {
  it("render correctly", () => {
    const smoothiesComponent = renderer.create(<Smoothies />).toJSON();
    expect(smoothiesComponent).toMatchSnapshot();
  });
});
