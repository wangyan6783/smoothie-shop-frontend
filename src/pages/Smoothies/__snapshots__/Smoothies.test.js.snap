// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Smoothies Component render correctly 1`] = `
<div
  className="smoothies-list"
/>
`;
